src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1933,13 +1933,7 @@ private BoundExpression SynthesizeReceiver(SyntaxNode node, Symbol member, Diagn
 
         internal Symbol ContainingMember()
         {
-            // We skip intervening lambdas and local functions to find the actual member.
-            var containingMember = this.ContainingMemberOrLambda;
-            while (containingMember.Kind != SymbolKind.NamedType && (object)containingMember.ContainingSymbol != null && containingMember.ContainingSymbol.Kind != SymbolKind.NamedType)
-            {
-                containingMember = containingMember.ContainingSymbol;
-            }
-            return containingMember;
+            return this.ContainingMemberOrLambda.ContainingNonLambdaMember();
```

Comment:
It seems like they end up doing the same thing in a different way:

- `Binder.ContainingMember()` works by traversing the `.ContainingSymbol` chain until `containingMember` or `containingMember.ContainingSymbol` is a type symbol, thus producing either a type or a member directly contained in a type
- `SymbolExtensions.ContainingNonLambdaMember(Symbol?)` works by going up the chain as long as `containingMember` is an anonymous function or local function. So if `containingMember` is a type or directly contained in a type it just returns the member.

I could be missing something, though, and since this change is not necessary to accomplish the goals of this PR, perhaps it should be removed from this PR or submitted in a separate PR.